server/routes/article.routes.js

```diff
@@ -1,19 +1,31 @@
 import express, { Router } from 'express';
-import { ArticleController, CommentController, ClapController } from '../controllers';
-import { tryCatch, createArticle, checkComments } from '../utils';
+import {
+  ArticleController,
+  CommentController,
+  ClapController
+} from '../controllers';
+import {
+  tryCatch,
+  createArticle as createarticle,
+  checkComments,
+  editarticle,
+  checkArticleExist
+} from '../utils';
 import { signInAuth } from '../utils/users/permissions.util';
 import imageUpload from '../config/cloudinaryconfig';
 
 const { createComments, threadedComment } = CommentController;
+const { createArticle, editArticle } = ArticleController;
 
 const router = new Router();
 
-router.post('/', [signInAuth, imageUpload, createArticle], tryCatch(ArticleController.createArticle));
+router.post('/', [signInAuth, imageUpload, createarticle], tryCatch(createArticle));
 
 router.post('/:slug/comment', [checkComments, signInAuth], tryCatch(createComments));
 
 router.post('/:slug/comment/:commentid/thread', [checkComments, signInAuth], tryCatch(threadedComment));
 
 router.post('/:articleId/claps', signInAuth, tryCatch(ClapController.createClap));
+router.put('/:slug', [signInAuth, checkArticleExist, editarticle], tryCatch(editArticle));
```

Comment:
Who can edit the article? All signed in users?

Reply:
@Chiazokam Thank you for pointing this to me, only the author of the article should be able to edit an article